doltpy_integrations/metaflow/examples/hospital_bounty/blog_post.md

```diff
@@ -0,0 +1,170 @@
+> As the observation count increases, the procedure level price variance should decrease
+
+Let's jump into defining a Metaflow pipeline that will answer this question.
+
+## Using Metaflow
+In this case we are using Dolt as the input, as well as for storing intermediate results and outputs. In many use-cases the upstream datasource might come from a data warehouse of some kind. The architecture for our flow looks something like:
+
+
+Our flow has three steps:
+- `start`: compute and store state level medians for every procedure
+- `variances`: compute procedure level variances across states
+- `end`: required by Metaflow
+
+In this case `start` reads and computes an intermediate result. This result is stored to Dolt. `variances` uses this intermediate result to compute the data of interest. The state level medians are not especially taxing to compute, but in a real world setting it's perfectly plausible that an intermediate result is used by many downstream jobs and cannot ergonomically be recomputed on the fly. Versioning provides read isolation to users of that intermediate result. This what the flow looks like schematically:
+![Flow definition schematic](dolt-metaflow-integration-sample-job-schematic.png)
+
+The Flow can be run as follows:
+```
+> poetry run python3 hospital_price_variance.py run \ 
+--hospital-price-db path/to/hospital-price-transparency
+--hospital-price-analysis-db path/to/hospital-price-analysis
+```
+
+```python
+    @step
+    def start(self):
+        read_conf = DoltConfig(database=self.hospital_price_db, branch=self.hospital_branch)
+        with DoltDT(run=self, config=read_conf) as dolt:
+            prices_by_state = """
+                SELECT
+                  h.state,
+                  p.code,
+                  p.payer,
+                  p.price
+                FROM
+                  prices p
+                  LEFT JOIN hospitals h ON p.npi_number = h.npi_number;
+            """
+            prices = dolt.sql(prices_by_state, as_key="prices")
+
+        median_price_by_state = prices.groupby(['state', 'code']).median()
+
+        write_conf = DoltConfig(database=self.hospital_price_analysis_db)
+        with DoltDT(run=self, config=write_conf) as dolt:
+            dolt.write(
+                median_price_by_state,
+                "state_procedure_medians",
+                ["state", "code"]
+            )
+
+    @step
+    def variances(self):
+        analysis_conf = DoltConfig(database=self.hospital_price_analysis_db, branch=self.hospital_branch)
+        with DoltDT(run=self, config=analysis_conf) as dolt:
+            median_price_by_state = dolt.read("state_procedure_medians")
+            variance_by_procedure = median_price_by_state.groupby("code").var()
+            dolt.write(variance_by_procedure, "variance_by_procedure")
+
+    @step
+    def end(self):
+        pass
+```
+
+Let's access that result via the Meatflow API to make clear the mechanism of the integration:
```

Comment:
Should definitely have called it Meatflow

Reply:
true... flows metaflow metadata data wtf